internal/external-dns/plan/plan.go

```diff
@@ -249,8 +280,14 @@ func (p *Plan) Calculate() *Plan {
 		if len(row.current) == 0 {
 			recordsByType := t.resolver.ResolveRecordTypes(key, row)
 			for _, records := range recordsByType {
-				if len(records.candidates) > 0 {
-					managedChanges.creates = append(managedChanges.creates, t.resolver.ResolveCreate(records.candidates))
+				candidates := []*endpoint.Endpoint{}
+				for _, c := range records.candidates {
+					if _, ok := c.Labels[SoftDeleteLabel]; !ok {
```

Comment:
I see that here we are checking for the presence of the label and later on we are checking for the value being `true`. Would `soft_delete=blah` break anything?

Reply:
yeah, it's a good point I was vacillating between needing a particular value, and just requiring the label to be present. I think expecting a value makes sense though. I'll update this.